sdk/include/opentelemetry/sdk/metrics/instruments.h

```diff
@@ -19,6 +19,7 @@ enum class InstrumentType
   kCounter,
   kHistogram,
   kUpDownCounter,
+  kGauge,
```

Comment:
Is inserting a new `InstrumentType` of `kGauge` a SDK breaking change and also break the ABI? Should it be included only for `OPENTELEMETRY_ABI_VERSION_NO >= 2`?

Reply:
It won't be ABI breaking, as this is not part of API surface. It could be breaking change for custom exporters which use the `switch` on this `enum` without handling `default` case, but this can be added in the changelog.

Reply:
This has been discussed before:

https://github.com/open-telemetry/opentelemetry-cpp/pull/3029#discussion_r1730141170

The code as written is fine:

* no change in the API, this kGauge is part of the SDK
* the SDK ABI is changed, and we do not guarantee SDK ABI compatibility

I don't think we should document that in the changelog for the SDK, by definition every SDK change can (and most of the time does) break the SDK ABI.

@ThomsonTan Please remove the change requested then, as this blocks the merge.